### PR TITLE
Adds `merge_existing_data` flag to `update_block_document` and new POST endpoint

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/PrefectHQ/prefect/issues/7264

I think this needs a follow-up on [UI to set `merge_existing_data=False`](https://github.com/PrefectHQ/orion-design/blob/main/src/services/WorkspaceBlockDocumentsApi.ts#L39-L41).

### Example

```
        await models.block_documents.update_block_document(
            session,
            block_document_id=block_document.id,
            block_document=schemas.actions.BlockDocumentUpdate(data=dict(x=2, merge_existing_data=False)),
        )
```

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: src/prefect/client/orion.py
Comment (madkinsz) [inline src/prefect/client/orion.py:984]: Why `replace` instead of `update`?

File: src/prefect/orion/schemas/actions.py
Comment (madkinsz) [inline src/prefect/orion/schemas/actions.py:318]: Similar to my comment above, our standard here is `Update` for all these actions, we should probably stick to that. We can document that it will not merge with existing data.

File: src/prefect/client/orion.py
Comment (ahuang11) [reply src/prefect/client/orion.py:984]: Based off https://github.com/PrefectHQ/prefect/issues/7264

When users try clearing the value from an optional field, it keeps the old value.

File: src/prefect/orion/schemas/actions.py
Comment (ahuang11) [reply src/prefect/orion/schemas/actions.py:318]: I think Alex wanted a separate endpoint for `post` so I copied `update_block_document` (which has `patch`) with `replace_block_document`. Then for consistency, I also added this model, but unsure if it's necessary.

File: src/prefect/client/orion.py
Comment (madkinsz) [reply src/prefect/client/orion.py:984]: Ah gee we have an existing `update` that uses PATCH. I see. I wonder if we should just make this a keyword argument for `update_block_document`? I'm hesitant to deviate from our existing naming patterns.

Comment (ahuang11) [reply src/prefect/client/orion.py:984]: If we make it a kwarg, what about the API?  https://github.com/PrefectHQ/prefect/pull/7470/files#diff-f5cd2ecfc6dc2faf3ae2463c457bfb363fd32636208d824c28bd6165dc25b0cbR138-R139

Comment (madkinsz) [reply src/prefect/client/orion.py:984]: We could call the same route with POST instead of PATCH. Just wondering, I don't see any differences from the existing method in the client / API implementations — is that work in progress still? Perhaps once I know what will need to change it'll be clearer.

Comment (ahuang11) [reply src/prefect/client/orion.py:984]: Oh I meant the naming of the function since you mentioned you're hesitant to deviate from our existing naming patterns.

Yeah there's no changes at the moment because I wasn't sure if changing `PATCH` to `POST` would automatically do everything for me, but based on your comment, I guess not. I probably have to add another `replace_block_document` function similar to [`update_block_document`](https://github.com/PrefectHQ/prefect/blob/e7e6808e056d1dd8ba9773073ca4c2b6bbc2bdf5/src/prefect/orion/models/block_documents.py#L421)

Comment (ahuang11) [reply src/prefect/client/orion.py:984]: Maybe I should use the PUT method instead based off https://stackoverflow.com/questions/31089221/what-is-the-difference-between-put-post-and-patch#:~:text=Here%20is%20a%20simple%20description,always%20for%20updating%20a%20resource

Comment (madkinsz) [reply src/prefect/client/orion.py:984]: Ah yeah POST/PATCH is just the type of request, we need to change our behavior accordingly.

I think in the models `update_block_document` can take a `merge_existing_data` bool. PATCH would set it to true and `POST` would set it to false. It's probably safe to reuse the `update_block_document` client method and `BlockDocumentUpdate` action.

Comment (madkinsz) [reply src/prefect/client/orion.py:984]: I think we may be using POST for cases like this elsewhere although a PUT may be semantically correct since we're only updating the `data` field. It's kind of a murky area. cc @zangell44 / @chrisguidry.

File: tests/orion/api/test_block_documents.py
Comment (desertaxle) [inline tests/orion/api/test_block_documents.py:983]: nit
```suggestion
class TestUpdateBlockDocumentWithoutMergingExistingData:
```

**Review by desertaxle — CHANGES_REQUESTED** [review]

**Comment by desertaxle** [comment]
Can you add a test that verifies that removing a block document reference works via the new PUT endpoint. I think it will work without any code changes, but having a test to verify that would be ideal.

**Comment by madkinsz** [comment]
Hoping to get some additional review from the backend team before merging. We'll need to bump Cloud for this as well.

File: src/prefect/orion/api/block_documents.py
Comment (abrookins) [inline src/prefect/orion/api/block_documents.py:140]: IMO, `replace_block_document_data` captures this better.

**Comment by abrookins** [comment]
Just making sure I follow these changes @madkinsz and sync up how I'd expect the HTTP request methods and naming to work:

- POST to **create** a new resource
- PATCH to **update** specific fields within an existing resource -- data could be a partial or complete representation of the resource
- PUT to **replace** an existing resource -- data must be a complete representation of the resource and will replace it

That's why I think think naming the API function `replace_...` makes the most sense.

File: src/prefect/orion/api/block_documents.py
Comment (zangell44) [inline src/prefect/orion/api/block_documents.py:139]: This is not the correct use of `PUT`, which should be "checking if resource exists then update, else create new resource".

**Review by zangell44 — CHANGES_REQUESTED** [review]
We should not mis-use PUT requests

File: src/prefect/orion/api/block_documents.py
Comment (desertaxle) [reply src/prefect/orion/api/block_documents.py:139]: Would using `POST` for this operation be more appropriate?

**Review by desertaxle — APPROVED** [review]
This LGTM, but I'm interested in hearing from @zangell44 if POST is the best HTTP operation to use for this functionality.

File: src/prefect/orion/models/block_documents.py
Comment (zangell44) [inline src/prefect/orion/models/block_documents.py:433]: I think we'll want to exclude the field `merge_existing_data` from the update values as well

**Review by zangell44 — APPROVED** [review]
One comment but this seems reasonable to me

**Review by desertaxle — APPROVED** [review]
LGTM!

**Comment by pleek91** [comment]
So using this we can just add `merge_existing_data: false` and that will replace all the data for a block with what is present in the request?

**Comment by ahuang11** [comment]
I believe so.

**Pull request merged**
